client/material/radio.tsx

```diff
@@ -150,23 +160,25 @@ export const RadioButton = React.memo(
     value,
     disabled,
     inputProps,
+    name,
     selected,
     onChange,
   }: RadioButtonProps<T>) => {
     const id = useId()
 
     const [buttonProps, rippleRef] = useButtonState({ disabled })
 
-    const handleChange = useStableCallback(() => {
+    const handleChange = useStableCallback((event: React.ChangeEvent<HTMLInputElement>) => {
```

Comment:
If you're going to pass the event through there's no particular reason for this to exist I don't think. In the props you pass to the input just do:
```ts
onChange: !disabled ? onChange : undefined
```

Reply:
Actually not even sure you need to be attaching the change handlers there at all though? May as well just attach it on the RadioGroup element and handle all the change events when they bubble there (the browser should properly prevent change events from happening on disabled inputs anyway)

Reply:
I did the first part of this, but when I do the second part, I get this error:
![image](https://github.com/ShieldBattery/ShieldBattery/assets/4757944/b6c1645d-77fe-4d88-a469-3a4b96f6624b)

Could probably be avoided using `defaultChecked` (even though I don't see that attribute documented anywhere?), but maybe just passing the `onChange` directly to the `input` is good as well?

Reply:
Possibly fine, alternatively you can just do one of:
1) declare a no-op function at the top-level in this file and pass that instead of `undefined`
2) do what it suggests and also set `readOnly` in the disabled case

Reply:
I couldn't really make the 2. work (I still can't find the documentation for `defaultChecked` anywhere), so I went with the 1. option.

Reply:
`defaultChecked` isn't relevant, that's just for uncontrolled components. That's a separate thing they're suggesting from setting `onChange` or `readOnly`.

If you set `defaultChecked`, the browser handles the checked/unchecked value from then on, just like if you set `defaultValue` on a text input. The error is saying that either your component needs to be uncontrolled, *or* you need to have one of: `onChange` or `readOnly` set.

I think solving it the way you did is fine, but just for future reference